cob_command_gui/src/knoeppkes.py

```diff
@@ -102,7 +102,24 @@ def start(func, args):
         largs.append("diff")
     #print "Args", tuple(largs)
     #print "func ", func
-    thread.start_new_thread(func,tuple(largs))
+    #thread.start_new_thread(func,tuple(largs))              # exits silently without evaluating result
+    thread.start_new_thread(call_thread,(func,tuple(largs))) # calls func(args) and shows result in a MessageDialog
+
+def call_thread(func,args):
+  result = func(*args)
+  if not result.success:    # action_handle returns with failure
+    result_dialog = gtk.MessageDialog(None, 0, gtk.MESSAGE_ERROR,
+                                               gtk.BUTTONS_OK,
+                                               "Executing " + func.__name__ + "(" + args[0] + ") failed!")
+    gtk.gdk.threads_enter()
+    result_dialog.format_secondary_text(result.message)
+    gtk.gdk.threads_leave()
+    gtk.gdk.threads_enter()
+    result_dialog.run()
+    gtk.gdk.threads_leave()
+    gtk.gdk.threads_enter()
+    result_dialog.destroy()
+    gtk.gdk.threads_leave()
```

Comment:
I'm not too familar either, but I had problems when __several__ pop-ups were opend (e.g. when pressing "INIT ALL") and this solution was proposed in a QA forum...